### PR TITLE
feat: (useridmapping) adds implements in-mem db functions and storage layer tests

### PR DESCRIPTION
**Pull request by jscyo**
## Summary of change
Adds in-mem db queries and storage layer tests for userId mapping

## Related issues
- https://github.com/supertokens/supertokens-core/pull/465

## Test Plan
- [ ] deleting a userId mapping with an unknown userid
- [ ] deleting a userId mapping
- [ ] updating externalUserId info using an unknown userid
- [ ] updating externalUserId

## Documentation changes
- none

## Checklist for important updates
- [ ] Changelog has been updated
    - [ ] If there are any db schema changes, mention those changes clearly
- [ ] `coreDriverInterfaceSupported.json` file has been updated (if needed)
- [ ] `pluginInterfaceSupported.json` file has been updated (if needed)
- [ ] Changes to the version if needed
   - In `build.gradle`
- [ ] Had installed and ran the pre-commit hook
- [ ] If there are new dependencies that have been added in `build.gradle`, please make sure to add them in `implementationDependencies.json`.
- [ ] Issue this PR against the latest non released version branch.
   - To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
   - If no such branch exists, then create one from the latest released branch.

## Remaining TODOs for this PR
- [ ] Item1
- [ ] Item2

### CODE REVIEW

File: src/test/java/io/supertokens/test/userIdMapping/UserIdMappingStorageTest.java [hunk 1]
```diff
@@ -341,4 +341,164 @@ public void testRetrievingUserIdMapping() throws Exception {
...
+
+        // create User
+        UserInfo userInfo = EmailPassword.signUp(process.main, "test@example.com", "testPass123");
+
+        String superTokensUserId = userInfo.id;
+        String externalUserId = "externalId";
+        String externalUserIdInfo = "externalUserIdInfo";
+
+        // create a userId mapping
+        storage.createUserIdMapping(superTokensUserId, externalUserId, externalUserIdInfo);
```
Comment (rishabhpoddar) [inline src/test/java/io/supertokens/test/userIdMapping/UserIdMappingStorageTest.java:464]: Add a different test where you are creating it with `null` as the externalUserIdInfo and see that when you retrieve it, it gives back null

**Review by rishabhpoddar — APPROVED** [review]
